src/main/groovy/edu/oregonstate/mist/trailsapi/db/TrailDAO.groovy

```diff
@@ -134,5 +134,37 @@ interface TrailDAO extends Closeable {
 		 LEFT JOIN TRAIL_DIFFICULTIES ON TRAILS.DIFFICULTY_ID = TRAIL_DIFFICULTIES.DIFFICULTY_ID
 		 WHERE TRAILS.ID = :id
 	""")
-	Trail getTrailByID(@Bind("id") Integer id)
+    Trail getTrailByID(@Bind("id") Integer id)
+
+    /***********************************************************************************************
+    PUT /trails/{trailID}
+    Updates values if binded value is not null
+    ***********************************************************************************************/
+    @SqlUpdate("""
+        UPDATE TRAILS
+        SET
+            NAME = COALESCE(:name, NAME),
+            DIFFICULTY_ID = COALESCE((SELECT DIFFICULTY_ID FROM TRAIL_DIFFICULTIES
```

Comment:
Are you doing any validation of the difficulty name when someone puts it in their request body? Given this query, it seems like if my difficulty doesn't correlate to a valid difficulty in the `TRAIL_DIFFICULTIES` table, it will just return the original difficulty associated with the trail.

Reply:
I will implement difficulty validation in `trailValidator` so that this is checked for POST and PUT